### PR TITLE
Warn if a package is known bad

### PR DESCRIPTION
**Pull request by dopsi**
Also, gracefully handle HTTP errors instead of crashing.

### CODE REVIEW

File: signoff/__init__.py [hunk 1]
```diff
@@ -200,6 +200,10 @@ def format_signoff_short(signoff_pkg, local_pkg, options):
     elif options.show_uninstalled and local_pkg:
         formatted += click.style(" (installed)", bold=True, fg="blue")
 
+    # show known bad indicator
+    if signoff_pkg["known_bad"]:
+        formatted += click.style(" (bad)", bold=True, fg="red")
```
Comment (haavard) [inline signoff/__init__.py]: Can we append ` (known bad)` here instead? It's not like we're tight on line length here, and it's more clear what it means.

File: signoff/__init__.py [hunk 2]
```diff
@@ -361,14 +380,20 @@ def main(action, uninstalled, signed_off, quiet, username, password, package,
...
+            warn_if_bad(signoff_pkg)
         if options.noconfirm or confirm("Sign off {}?".format(
                 click.style(" ".join(pkgbases), bold=True))):
             for signoff_pkg, local_pkg in packages:
-                session.signoff_package(signoff_pkg)
-                click.echo("Signed off {}.".format(signoff_pkg["pkgbase"]))
+                try:
+                    session.signoff_package(signoff_pkg)
+                except requests.exceptions.HTTPError as e:
+                    click.echo("Could not sign off {} ({})".format(signoff_pkg["pkgbase"], e))
```
Comment (haavard) [inline signoff/__init__.py]: What does an example error message look like here? I'm not able to test this myself right now.

Comment (dopsi) [reply signoff/__init__.py]: It looks like

   Could not sign off <package> (403 Client Error: Forbidden for url: https://www.archlinux.org/packages/<repo>/x86_64/<package>/signoff/)

File: signoff/__init__.py (hunk 1 above)
Comment (dopsi) [reply signoff/__init__.py]: Fixed in b74554b

File: signoff/__init__.py (hunk 2 above)
Comment (haavard) [reply signoff/__init__.py]: Looks good to me!

**Comment by haavard** [comment]
@jelly, can you take a look and merge?

**Comment by jelly** [comment]
Merged via Git.

**Pull request closed without merging**